### DISCUSSION TITLE
WARNING: Crazyflie USB: No backend available

### DISCUSSION THREAD

**Discussion by ClaesJensen** [bitcraze/discussions#1626] [Q&A]
Hi everyone!
I am trying to setup my cfclient to connect with the crazyradio in order to communicate with my crazyflie 2.1 drone. However, when I try to scan for an interface, this occurs in my terminal:
INFO:cflib.drivers.cfusb:Looking for devices....
WARNING:cflib.crtp.usbdriver:Exception while scanning for Crazyflie USB: No backend available
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

Upon boot the messages are as shown here:
INFO:__main__:Disabling STL printouts
INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
INFO:cfclient.utils.input.inputreaders:Successfully initialized [linuxjsdev]
INFO:cfclient.utils.input.inputreaders:Could not initialize [pysdl2]: No SDL2 support on Linux
INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
INFO:cfclient.utils.config:Dist config read from /snap/cfclient/54/lib/python3.6/site-packages/cfclient/configs/config.json
INFO:cfclient.utils.config:Config file read from [/home/claes/snap/cfclient/54/.config/cfclient/config.json]
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file
INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file
INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file
Qt: Session management error: Could not open network socket
INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cfclient.ui.tabs.QualisysTab:Switching Flight Mode to: FlightModeStates.DISCONNECTED
WARNING:cflib.crtp.usbdriver:Exception while scanning for Crazyflie USB: No backend available
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
pci id for fd 20: 8086:9a49, driver (null)
libGL error: failed to create dri screen
libGL error: failed to load driver: iris

I have tried on both my desktop windows PC and my linux ubuntu laptop (the one where these terminal messages are from).
As far as versions go i have:
Python 3.10.12
Ubuntu 22.04.5 LTS x86_64

If any other information is needed to help me troubleshoot this issue, please reach out and i will get it to you ASAP.

I have tried to search for people with a similar issue, but their solutions have not seemed to help me... :)

Best regards

**Comment by gemenerik** [comment]
The error you’re seeing when trying to connect cfclient to Crazyradio is often due to missing dependencies or configuration issues with the USB libraries on Linux.

- Ensure that your user has the correct permissions to access USB devices. On Ubuntu, this usually involves setting up udev rules.
- Verify PyUSB (pip) and LibUSB (ubuntu) Installation
